src/wallet/walletdb.cpp

```diff
@@ -847,6 +855,45 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
                 return false;
             }
         }
+        else if (strType == "recipientmapping")
+        {
+            uint256 txid;
+            CSerializeRecipientAddress recipient;
+            std::string rawUa;
+            ssKey >> txid;
+            ssKey >> recipient;
+            ssValue >> rawUa;
+
+            auto pa = keyIO.DecodePaymentAddress(rawUa);
```

Comment:
This is clunky. I think that we should factor out the contents of `KeyIO::DecodePaymentAddress` here to a static method on `UnifiedAddress` that takes the key constants as an argument.